common/common.cpp

```diff
@@ -886,6 +900,48 @@ std::string gpt_random_prompt(std::mt19937 & rng) {
     GGML_UNREACHABLE();
 }
 
+//
+// String parsing
+//
+
+std::string parse_samplers_input(std::string input){
+    std::string output = "";
+    // since samplers names are written multiple ways
+    // make it ready for both system names and input names
+    std::unordered_map<std::string, char> samplers_symbols{
+        {"top_k",      'k'},
+        {"top-k",      'k'},
+        {"top_p",      'p'},
+        {"top-p",      'p'},
+        {"nucleus",    'p'},
+        {"typical_p",  'y'},
+        {"typical-p",  'y'},
+        {"typical",    'y'},
+        {"min_p",      'm'},
+        {"min-p",      'm'},
+        {"tfs_z",      'f'},
+        {"tfs-z",      'f'},
+        {"tfs",        'f'},
+        {"temp",       't'},
+        {"temperature",'t'}
+    };
+    // expected format example: "temp;top_k;tfs_z;typical_p;top_p;min_p"
+    size_t separator = input.find(';');
+    while (separator != input.npos){
+        std::string name = input.substr(0,separator);
+        input = input.substr(separator+1);
+        separator = input.find(';');
+        
+        if (samplers_symbols.find(name) != samplers_symbols.end()){
+            output += samplers_symbols[name];
+        }
+    }
+    if (samplers_symbols.find(input) != samplers_symbols.end()){
+        output += samplers_symbols[input];
+    }
+    return output;
+}
```

Comment:
```suggestion
std::string parse_samplers_input(std::string input) {
    std::string output = "";
    // since samplers names are written multiple ways
    // make it ready for both system names and input names
    std::unordered_map<std::string, char> samplers_symbols {
        {"top_k",      'k'},
        {"top-k",      'k'},
        {"top_p",      'p'},
        {"top-p",      'p'},
        {"nucleus",    'p'},
        {"typical_p",  'y'},
        {"typical-p",  'y'},
        {"typical",    'y'},
        {"min_p",      'm'},
        {"min-p",      'm'},
        {"tfs_z",      'f'},
        {"tfs-z",      'f'},
        {"tfs",        'f'},
        {"temp",       't'},
        {"temperature",'t'}
    };
    // expected format example: "temp;top_k;tfs_z;typical_p;top_p;min_p"
    size_t separator = input.find(';');
    while (separator != input.npos) {
        std::string name = input.substr(0,separator);
        input = input.substr(separator+1);
        separator = input.find(';');
        
        if (samplers_symbols.find(name) != samplers_symbols.end()) {
            output += samplers_symbols[name];
        }
    }
    if (samplers_symbols.find(input) != samplers_symbols.end()) {
        output += samplers_symbols[input];
    }
    return output;
}
```